assets/js/blocks/cart-checkout/cart/index.js

```diff
@@ -36,35 +34,27 @@ const settings = {
 		multiple: false,
 	},
 	example,
-	attributes: {
-		isShippingCalculatorEnabled: {
-			type: 'boolean',
-			default: IS_SHIPPING_CALCULATOR_ENABLED,
-		},
-		isShippingCostHidden: {
-			type: 'boolean',
-			default: IS_SHIPPING_COST_HIDDEN,
-		},
-	},
-
+	attributes: blockAttributes,
 	edit,
 
 	/**
 	 * Save the props to post content.
 	 */
 	save( { attributes } ) {
-		const {
-			className,
-			isShippingCalculatorEnabled,
-			isShippingCostHidden,
-		} = attributes;
-		const data = {
-			'data-is-shipping-calculator-enabled': isShippingCalculatorEnabled,
-			'data-is-shipping-cost-hidden': isShippingCostHidden,
-		};
+		const data = {};
+
+		Object.keys( blockAttributes ).forEach( ( key ) => {
+			if (
+				blockAttributes[ key ].save !== false &&
+				typeof attributes[ key ] !== 'undefined'
+			) {
```

Comment:
whar are we guarding against?

Reply:
This pattern (used in some other blocks too) allows us to prevent attributes saving to markup. This is just copy-pasted from other blocks.